ostd/src/mm/page/mod.rs

```diff
@@ -52,24 +53,32 @@ impl<M: PageMeta> Page<M> {
     ///
     /// The function panics if:
     ///  - the physical address is out of bound or not aligned;
-    ///  - the page is already in use.
+    ///  - the page is already in use;
+    ///  - the metadata's size exceeds [`PAGE_METADATA_MAX_SIZE`];
+    ///  - the alignment of the metadata exceeds [`PAGE_METADATA_MAX_ALIGN`].
     pub fn from_unused(paddr: Paddr, metadata: M) -> Self {
         assert!(paddr % PAGE_SIZE == 0);
         assert!(paddr < MAX_PADDR.load(Ordering::Relaxed) as Paddr);
+
+        // Checking unsafe preconditions of the `PageMeta` trait.
+        assert!(size_of::<M>() <= PAGE_METADATA_MAX_SIZE);
+        assert!(align_of::<M>() <= PAGE_METADATA_MAX_ALIGN);
```

Comment:
I think these conditions have already been guaranteed by the `PageMeta` trait.

Reply:
Since the unsafe trait `PageMeta` still can be implemented unsafely without checking. Some sanity check here is still necessary. I've turned them into debug assertions.

Reply:
Ok